backend/baseclass.pm

```diff
@@ -1207,8 +1207,10 @@ sub new_ssh_connection ($self, %args) {
     # timeout requires libssh2 >= 1.2.9 so not all versions might have it
     my $ssh = Net::SSH2->new(timeout => ($bmwqemu::vars{SSH_COMMAND_TIMEOUT_S} // 5 * ONE_MINUTE) * 1000);
 
-    # Retry multiple times, in case of the guest is not running yet
-    my $counter = $bmwqemu::vars{SSH_CONNECT_RETRY} // 5;
+    # Retry multiple times in case the guest is not running yet
+    my $counter = $bmwqemu::vars{SSH_CONNECT_RETRY} // 7;
+    my $interval = $bmwqemu::vars{SSH_CONNECT_RETRY_INTERVAL} // 10;
+    my $factor = $bmwqemu::vars{SSH_CONNECT_RETRY_FACTOR} // 2;
```

Comment:
This combination does not make sense to me. If the "guest is not running yet" then it's more likely to be up the later we try. So rather than exponentially increasing backoff it could be the inverse. To keep our change simple I am reconsidering: How about you keep your change about making the sleep interval more explicit and in a separate commit increase the retry, but don't introduce the factor

Reply:
During the estimation we discussed that the test fails because the machine takes longer to come up. In these situations we use exponential backoff.

Reply:
Alternatively there's a suggestion to simply increase retries for this test scenario. Maybe we don't want a generic solution afterall.

Reply:
> During the estimation we discussed that the test fails because the machine takes longer to come up. In these situations we use exponential backoff.
Well, we use it in case of network problems. Like if something fails try immediately, if it still does not work then maybe something is unavailable for longer in the network. In some cases we cover over potential outages of hosts. But here it's different because the host we want to reach was triggered for reboot by the very same execution flow. So that's making it more likely that a host is reachable the later you are in the waiting process unlike in other cases where we don't know why we can't reach a remote end.

> Alternatively there's a suggestion to simply increase retries for this test scenario. Maybe we don't want a generic solution afterall.
yes, that's what I suggested to do but in a separate commit. If you prefer you can also just bump the retry and revert all other changes although I find your change to make the sleep period explicit a good idea.

Reply:
Making it exponential is maybe really not so helpful here. In addition to what @okurz said, I think the tries here are also pretty cheap - so it isn't really problematic to do many frequent retries.

Reply:
Not sure about that separate commit. I assume the scenario-specific retry will need to be done in a job template somewhere.

Reply:
I guess that would be https://openqa.suse.de/group_overview/131

Reply:
https://gitlab.suse.de/qac/qac-openqa-yaml/-/merge_requests/1703